vendor/k8s.io/kubernetes/pkg/kubectl/resource_printer_test.go

```diff
@@ -241,9 +263,44 @@ func TestCustomTypePrinting(t *testing.T) {
 	}
 }
 
+func TestCustomTypePrintingWithKind(t *testing.T) {
+	columns := []string{"Data"}
+	printer := NewHumanReadablePrinter(&PrintOptions{WithKind: true, Kind: "test"})
+	printer.Handler(columns, PrintCustomType)
+
+	obj := TestPrintType{"test object"}
+	buffer := &bytes.Buffer{}
+	err := printer.PrintObj(&obj, buffer)
+	if err != nil {
+		t.Fatalf("An error occurred printing the custom type: %#v", err)
+	}
+	expectedOutput := "Data\ntest/test object"
+	if buffer.String() != expectedOutput {
+		t.Errorf("The data was not printed as expected. Expected:\n%s\nGot:\n%s", expectedOutput, buffer.String())
+	}
+}
+
+func TestCustomTypePrintingWithForcedKind(t *testing.T) {
+	columns := []string{"Data"}
+	printer := NewHumanReadablePrinter(&PrintOptions{WithKind: false})
+	printer.EnsurePrintWithKind("test")
```

Comment:
We should not call this method in the test.
What we should be testing is that when the input contains multiple types, then the printer will automatically include the kinds.

If the test calls `EnsurePrintWithKind` it is cheating.